src/platform/Zephyr/ConfigurationManagerImpl.cpp

```diff
@@ -176,6 +176,7 @@ void ConfigurationManagerImpl::RunConfigUnitTest(void)
 void ConfigurationManagerImpl::DoFactoryReset(intptr_t arg)
 {
     ChipLogProgress(DeviceLayer, "Performing factory reset");
+    ThreadStackMgr().LockThreadStack();
```

Comment:
This must be guarded with  `#ifdef CONFIG_NET_L2_OPENTHREAD`
```suggestion
#ifdef CONFIG_NET_L2_OPENTHREAD
    ThreadStackMgr().LockThreadStack();
#endif
```
Also, as Damian suggested, please describe what is the purpose of this call here and why it can be used here without unlocking the stack.